### ISSUE TITLE
Playing audio directly from the string

### ISSUE THREAD

**Issue by debu-rnpr** [beeping-io/sdk-android#3]
Is there any way to play the audio directly from a string or byte[] instead of .wav file?

**observation** - while going through the beeping-core source code I found this method _Java_com_beeping_AndroidBeepingCore_BeepingCoreJNI_playBeeping(JNIEnv *env, jobject obj, jstring stringToEncode, jlong beepingObject)_, I created a corresponding native(from java code) function and call it, but in the logs I am getting BC_TOKEN_END_BAD(2)

**Comment by unic0x** [comment]
You will need to patch CoreJNI and expose as a native method and then in BeepingCore.java you will need to make another method that takes a string and pass this to the native method.

**Comment by debu-rnpr** [comment]
Exactly thats how I implemented, created a native function in BeepingCoreJNI.java and then called that function from BeepingCore.java

**Comment by alfredrc** [comment]
Hello @debu-rnpr ,

So, if you are done with, could you create the pull requests to upload the new source code ?

Thank you !!

**Comment by unic0x** [comment]
Let me know if you require any PR on this issue also worth exposing the play audio signature method.

**Comment by debu-rnpr** [comment]
Hello @alfredrc 
Here are the changes that I made

**BeepingCoreJNI.java -** 
    public final native int playBeeping(String stringToPlay,long beepingObject);

**BeepingCore.java -**
    public void playString(String s) {
        mBeepingCoreJNI.playBeeping(s, mBeepingObject);
    }

For testing Created an activity
**MainActivity.java -**
        // Init Beeping Framework
        BeepingCore beeping = new BeepingCore(this);
        // Listening
        beeping.startBeepingListen();
        setContentView(R.layout.activity_main);
        findViewById(R.id.play).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                beeping.playString("12345");
            }
        });

Do you need a PR for that?

**Comment by alfredrc** [comment]
Yes please, it's an open source project and the way to work. You have improved the source code and it's the way to go forward. So, waiting for you PR ... 

After that I will update the official docs.

I will do the same changes in the iPhone SDK.

Thank You !!!

**Comment by debu-rnpr** [comment]
But the issue is still there, I am still getting BC_TOKEN_END_BAD

**Comment by alfredrc** [comment]
Ok, I understand you trying to send from your Smartphone the Beep using the play method, and other device is not able to recognize the beep ... Is it right ?

**Comment by debu-rnpr** [comment]
Folllowing are my test scenarios - 

1. One device is playing the sound and the other one is listening(both android)
2. Single android device is both playing and listening 

In both the cases I am using INAUDIBLE(i.e configure(EnumBeepingMode.MODE_NONAUDIBLE))

**Comment by alfredrc** [comment]
Ok, I'll do check

**Comment by debu-rnpr** [comment]
Hi @alfredrc any update on this?

**Comment by alfredrc** [comment]
Working on that !

**Comment by unic0x** [comment]
> But the issue is still there, I am still getting BC_TOKEN_END_BAD

The reason for this issue is that the SDK 9 characters to be transmitted the last for are time.
